src/Event.hpp

```diff
@@ -68,6 +98,22 @@ class Event {
         return size;
     }
 
+    std::bitset<bitset_size> generate_comparison_set(
+                                                    std::uint32_t timestamp,
+                                                    std::uint32_t send_time,
+                                                    std::string sender_name,
+                                                    std::uint64_t generation,
+                                                    EventType event_type 
+                                                    )
+    {
+       const std::bitset<64U> b_timestamp(timestamp);
+       const std::bitset<64U> b_send_time(send_time);
+       const std::bitset<64U> b_sender_name(std::hash<std::string>{}(sender_name));
+       const std::bitset<1U> b_generation(generation);
+       const std::bitset<1U> b_event_type(static_cast<bool>(event_type));
+       return concatLoop<194, 64, 64, 64, 1, 1>(b_timestamp,b_send_time,b_sender_name,b_generation,b_event_type);
+    }
+
```

Comment:
Line 50, i dont get it how the values are passed from derived class to here. never seen this kind of assignment. If you can explain that to me it would be great. 
I am temporarily using the standard hash function on line  111, from C++ library, will change it when other logic is sound.